Make May 2025 news public, add June template (#30594)
idaholab/moose: modules/doc/content/newsletter/2025/2025_05.md

# MOOSE Newsletter (May 2025)

## MOOSE Improvements

> ### MFEM capabilty development

Reviewer: ```suggestion
### MFEM capability development
```

- The [FunctorExtremaPositions.md] was added to MOOSE to gather the locations of N extrema (max/min/max absolute) of a functor. In future, this will let us spawn simulations at these bounding locations.
- [Physics](Physics/index.md) can now skip object-creation tasks through the `dont_create_kernels/bcs/...` parameters that a user passes. This can be useful to combine `Physics` with standard MOOSE syntax.
- Recover testing has been enabled on the `contact` and `combined` modules. This ensures that future development in those
  modules can either create a checkpoint and recover from this checkpoint, or properly error if it does not support this
  feature.
- PETSc options can now be passed to each equation solve inside SubChannel problems, using the pre-defined prefixes.
- An additional `EXEC_POST_ADAPTIVITY` execution schedule was added to MOOSE. This notably enables transferring markers to match adaptivity refinement patterns between a parent and its child applications.

> - The Makefile for moose was improved to avoid errors on missing libtorch or libmfm on unrelated targets.

Reviewer: ```suggestion
- The Makefile for moose was improved to avoid errors on missing libtorch or libmfem on unrelated targets.
```

- JSON output of the perf graph was enabled. This is intended for automated examination of solver performance by the MOOSE [continuous integration platform](https://civet.inl.gov).
- The [ADEqualValueEmbeddedConstraint.md] was added to MOOSE. This facilitates the use of automatic differentiation (AD) in that system, enabling numerically-perfect Jacobians without hand-coding the Jacobian.
- The [LibtorchNeuralNetControl.md] object can now be used for recover simulations.
- The displaced mesh is now correctly displaced if adaptivity is run without actually modifying the mesh, because the `Markers` showed no adaptivity was needed.

> - The reporting of the preconditioner in the simulaiton header has been improved to correctly report the preconditioner when modified using the command line.

Reviewer: ```suggestion
- The reporting of the preconditioner in the simulation header has been improved to correctly report the preconditioner when modified using the command line. Reporting for simulations with multiple systems/variables has also been made clearer.
```
